Source/CarthageKit/Xcode.swift

```diff
@@ -474,6 +508,17 @@ private func mergeBuildProducts(
 					)
 				}
 
+			let mergeProductSwiftHeaderFilesIfNeeded = SignalProducer.zip(simulatorBuildSettings.executableURL, deviceBuildSettings.executableURL, outputURL)
+				.flatMap(.concat) { (simulatorURL: URL, deviceURL: URL, outputURL:  URL) -> SignalProducer<(), CarthageError> in
+					guard isSwiftFramework(productURL) else { return .empty }
```

Comment:
I'm onboard with the current solution of only header-merging `swiftmodule`-containing frameworks, but we might consider widening this to all frameworks once we have more time to gather data on if that's mostly safe or less-so.

We've [kinda been given indications](https://twitter.com/UINT_MIN/status/1100085577158815755) that differing of headers might also happen under some non-Swift-framework criteria via the new build system.

I've not actually seen (or seen indications) that these non-Swift-but-still-mostly-vanilla frameworks that produce differing headers might already be out there — but if we get reports of them, we should put out another release that drops this `isSwiftFramework` (in some form) and accommodates them.